### ISSUE TITLE
[Bug]: LLM.generate() collapse with some padding side

### ISSUE THREAD

**Issue by kevin3314** [vllm-project/vllm#5236]
### Your current environment

```text
Versions of relevant libraries:
[pip3] numpy==1.26.4
[pip3] nvidia-nccl-cu12==2.20.5
[pip3] torch==2.3.0
[pip3] triton==2.3.0
[pip3] vllm_nccl_cu12==2.18.1.0.4.0
[conda] Could not collectROCM Version: Could not collect
Neuron SDK Version: N/A
vLLM Version: 0.4.2
```

### 🐛 Describe the bug

Padding the input with some side seems to causes the generation by the LLM to break down. In the examples, I have only run this for one model, but the same phenomenon occurs with other models as well.
Reasonable generation results can be obtained by not padding the input, but I would greatly appreciate it if you could explain the cause of this issue.

```python
from __future__ import annotations

from contextlib import contextmanager

from transformers import AutoTokenizer, PreTrainedTokenizer
from vllm import LLM, SamplingParams

model_name = "Qwen/Qwen1.5-7B-Chat"
tokenizer: PreTrainedTokenizer = AutoTokenizer.from_pretrained(model_name)
llm = LLM(model_name)
print("Original padding side: ", tokenizer.padding_side)

prompt1 = "Give me a short introduction to large language model."
# prompt2 is short, so padded.
prompt2 = "Hello"
messages = [
    [{"role": "system", "content": "You are a helpful assistant."}, {"role": "user", "content": prompt1}],
    [{"role": "system", "content": "You are a helpful assistant."}, {"role": "user", "content": prompt2}],
]
text = tokenizer.apply_chat_template(messages, tokenize=False, add_generation_prompt=True)

@contextmanager
def set_temporal_padding_side(tokenizer: PreTrainedTokenizer, padding_side: str):
    """Temporarily set the padding side of the tokenizer.

    Useful when the padding side of the tokenizer is unknown or needs to be overridden.
    """
    original_padding_side = tokenizer.padding_side
    tokenizer.padding_side = padding_side
    try:
        yield
    finally:
        tokenizer.padding_side = original_padding_side

# generation with padding on given side.
def generate_with_padding_side_specified(side):
    with set_temporal_padding_side(tokenizer, side):
        model_inputs = tokenizer(
            text,
            return_tensors="pt",
            padding=True,
        )

    vllm_outputs = llm.generate(
        prompt_token_ids=model_inputs.input_ids.tolist(),
        sampling_params=SamplingParams(),
        use_tqdm=False,
    )
    return tokenizer.decode(vllm_outputs[1].outputs[0].token_ids)


left_pad_generated_texts = generate_with_padding_side_specified("left")
right_pad_generated_texts = generate_with_padding_side_specified("right")

print("Output with left padding:", left_pad_generated_texts)
print("Output with right padding:", right_pad_generated_texts)
```

I got the following output:
```
Original padding side:  right
Output with left padding: Hello! How can I assist you today?<|im_end|>
Output with right padding: 圜 diplomácii byly posíleny každoročn
```

**Comment by mgoin** [comment]
Hi @kevin3314 it is not necessary to pad the inputs because vLLM can handle batches with different lengths with continuous batching. When you are producing padding on the right-side and passing in the `prompt_token_ids` directly, there is no context for vLLM that the token_ids have padding and thus should have an edited attention_mask. So it will simply try to continue generating using those padding tokens as being part of the prompt. 
Because of this, I would recommend simply passing a list of tokenized prompts without them being padded. vLLM will be able to batch them together.
